Wrappers/Python/test/test_DataContainer.py

```diff
@@ -781,8 +786,103 @@ def test_reduction_mean(self):
         mean = data.mean()
         expected = numpy.float64(0+1+2+3)/numpy.float64(4)
         numpy.testing.assert_almost_equal(mean, expected)
+
+
+    def directional_reduction_unary_test(self, data, test_func, expected_func, out):
+        # test specifying mean in 1 axis
+        result = test_func(direction=data.dimension_labels[1])
+        numpy.testing.assert_almost_equal(result.as_array(), expected_func(data.as_array(), axis=1))
+        numpy.testing.assert_equal(result.dimension_labels,(data.dimension_labels[0],data.dimension_labels[2]))
+        result = test_func(axis=1)
+        numpy.testing.assert_almost_equal(result, expected_func(data.as_array(), axis=1))
+        # test specifying mean in 2 axes
+        result = test_func(direction=(data.dimension_labels[0],data.dimension_labels[1]))
+        numpy.testing.assert_almost_equal(result.as_array(), expected_func(data.as_array(), axis=(0,1)))
+        numpy.testing.assert_equal(result.dimension_labels,(data.dimension_labels[2],))
+        # test specifying mean in 3 axes
+        result = test_func(direction=(data.dimension_labels[0],data.dimension_labels[1],data.dimension_labels[2]))
+        numpy.testing.assert_almost_equal(result, expected_func(data.as_array()))
+        # test specifying direction with an int   
+        with numpy.testing.assert_raises(ValueError):
+            result = test_func(direction=0)
```

Comment:
Is this what we want to happen? @paskino mentioned sirf compatibility might suggest it could take and axis index too.